zerver/lib/streams.py

```diff
@@ -511,17 +520,28 @@ def list_to_streams(streams_raw: Iterable[StreamDict],
     """
     # Validate all streams, getting extant ones, then get-or-creating the rest.
 
-    stream_set = {stream_dict["name"] for stream_dict in streams_raw}
+    stream_set = {stream_dict["stream"] for stream_dict in streams_raw}
+    is_stream_type_name = isinstance(next(iter(stream_set), None), str)
+
+    if is_stream_type_name:
+        for stream_name in stream_set:
+            # Stream names should already have been stripped by the
+            # caller, but it makes sense to verify anyway.
+            assert isinstance(stream_name, str) and stream_name == stream_name.strip()
+            check_stream_name(stream_name)
 
-    for stream_name in stream_set:
-        # Stream names should already have been stripped by the
-        # caller, but it makes sense to verify anyway.
-        assert stream_name == stream_name.strip()
-        check_stream_name(stream_name)
+    else:
+        for stream_id in stream_set:
+            # Stream id should already be parsed in int by the caller,
```

Comment:
All `dict`s in `streams_raw` must either have the name or the id as `stream` paramter correct? In that case, I think we should specify that clearly in the docstring of `list_to_streams`

Reply:
Also, FWIW how about using a more descriptive name for the `StreamDict` paramter than just `stream`?

Reply:
What do you think will be a better name for it?

Reply:
My suggestion: `stream`, `stream_identifer`.

Reply:
Hmm, I'm not really sure what a good name for it could be. Anything involving "identifier" could be confused with `id`.
I pointed this out because `stream` is also used to indicate the name in other parts of the codebase.
Maybe it would be best to somehow have both  `id` and `name` as keys in `StreamDict`?

Reply:
Actually, Tim suggested that there should be only one parameter to represent both the stream name and stream id.
(https://github.com/zulip/zulip/issues/10744#issuecomment-434875243).
IDK If it helps but I have added a comment in `StreamDict` that the `stream` attribute is used to represent both.

Reply:
I think if we're passing a dict, we should use different parameter names; I had been thinking about a "list" context.

Reply:
@timabbott  Can you please explain a bit about what `list context` means?

Reply:
I'd been imagining a list of things that are either a string or an integer, not a dictionary.